src/arbOwnerPrepareTransactionRequest.ts

```diff
@@ -5,45 +5,57 @@ import {
   Address,
   Chain,
   Transport,
+  PrepareTransactionRequestReturnType,
+  EncodeFunctionDataReturnType,
+  ContractFunctionArgs,
 } from 'viem';
 
 import { arbOwner } from './contracts';
 import { upgradeExecutorEncodeFunctionData } from './upgradeExecutor';
-import { GetFunctionName } from './types/utils';
+import { GetPrepareTransactionRequestParams } from './types/utils';
 
 type ArbOwnerAbi = typeof arbOwner.abi;
-export type ArbOwnerPrepareTransactionRequestFunctionName = GetFunctionName<ArbOwnerAbi>;
-export type ArbOwnerEncodeFunctionDataParameters<
-  TFunctionName extends ArbOwnerPrepareTransactionRequestFunctionName,
-> = EncodeFunctionDataParameters<ArbOwnerAbi, TFunctionName>;
+export type ArbOwnerFunctionName = GetPrepareTransactionRequestParams<ArbOwnerAbi>;
+type ArbOwnerPrepareTransactionRequestArgs<TFunctionName extends ArbOwnerFunctionName> =
+  ContractFunctionArgs<ArbOwnerAbi, 'nonpayable' | 'payable', TFunctionName>;
 
-function arbOwnerEncodeFunctionData<
-  TFunctionName extends ArbOwnerPrepareTransactionRequestFunctionName,
->({ functionName, abi, args }: ArbOwnerEncodeFunctionDataParameters<TFunctionName>) {
+type ArbOwnerEncodeFunctionDataParameters<
+  TFunctionName extends ArbOwnerFunctionName,
+  TArgs extends ArbOwnerPrepareTransactionRequestArgs<TFunctionName> = ArbOwnerPrepareTransactionRequestArgs<TFunctionName>,
+> = EncodeFunctionDataParameters<ArbOwnerAbi, TFunctionName> & {
+  args: TArgs;
+};
+
+function arbOwnerEncodeFunctionData<TFunctionName extends ArbOwnerFunctionName>({
+  functionName,
+  abi,
+  args,
+}: ArbOwnerEncodeFunctionDataParameters<TFunctionName>): EncodeFunctionDataReturnType {
   return encodeFunctionData({
     abi,
     functionName,
     args,
-  });
+  } as EncodeFunctionDataParameters);
 }
```

Comment:
Applies to all `xxxEncodeFunctionData`: we could probably remove them and directly call `encodeFunctionData(params)` in `xxxprepareFunctionData`. It would simplify type